From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Anderson <thomasanderson@chromium.org>
Date: Mon, 17 Aug 2026 17:55:49 -0700
Subject: [M151] [ui/gtk] Pin gtk-modules to an empty string during
 initialization

Original change's description:
> [ui/gtk] Pin gtk-modules to an empty string during initialization
>
> Setting `gtk-modules` to `""` in GtkUi::Initialize() assigns
> GTK_SETTINGS_SOURCE_APPLICATION priority to the setting, preventing
> lower-priority XSETTINGS updates from overriding `gtk-modules` and
> executing untrusted GTK modules in the unsandboxed browser process via a
> retained X11 connection.
>
> Fixed: 545124048
> Change-Id: I91e1c3df94c0567c69eec676da26ee1bb46eca2f
> Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8243490
> Commit-Queue: Thomas Anderson <thomasanderson@chromium.org>
> Reviewed-by: Lei Zhang <thestig@chromium.org>
> Cr-Commit-Position: refs/heads/main@{#1678319}

(cherry picked from commit 9ec61d5e14fff874426e560d593f93094c4377b7)

Bug: 546660197,545124048
Change-Id: I91e1c3df94c0567c69eec676da26ee1bb46eca2f
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8258459
Commit-Queue: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Auto-Submit: chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com <chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com>
Cr-Commit-Position: refs/branch-heads/7922@{#3246}
Cr-Branched-From: 059c884787b1bc15faa5afe5250f7d3a7e0ef64a-refs/heads/main@{#1654411}

diff --git a/ui/gtk/gtk_ui.cc b/ui/gtk/gtk_ui.cc
index 0bc31770336d43e41cd8d783fa77835019573ef4..7a18d6a7279c00b093d343a16eb6a406fda6833a 100644
--- a/ui/gtk/gtk_ui.cc
+++ b/ui/gtk/gtk_ui.cc
@@ -366,6 +366,9 @@ bool GtkUi::Initialize() {
   };
 
   GtkSettings* settings = gtk_settings_get_default();
+  // Pin `gtk-modules` to an empty string with APPLICATION source priority
+  // to prevent XSETTINGS updates from loading GTK modules.
+  g_object_set(settings, "gtk-modules", "", nullptr);
   SanitizeIconThemeName();
   SanitizeThemeName();
   SanitizeCursorThemeName();
